smolyay/samples.py

```diff
@@ -0,0 +1,409 @@
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points within a domain.
+    """
+
+    def __init__(self, domain):
+        self._points = None
+        self._domain = numpy.array(domain)
+        self._valid_cache = False
+
+    @property
+    def domain(self):
+        """numpy.ndarray: Domain of the `points`."""
+        return self._domain
+
+    @property
+    def points(self):
+        """numpy.ndarray: Points in the set."""
+        if not self._valid_cache:
+            self._create()
+        self._valid_cache = True
+        return self._points
+
+    @abc.abstractmethod
+    def _create(self):
+        """Create the points in the set."""
+        pass
+
+    @staticmethod
+    def scale_to_domain(x, old, new):
```

Comment:
I think this should be a normal method so that it can access the domain. I think it should also be private, since I don't think we want it to be part of the public API in case we change our mind.
```suggestion
    def _scale_to_domain(self, points, old_domain):
```
Last, please remove print statements used for debugging.